test/spiflash_stress_test/src/spiflash_stress_test.c

```diff
@@ -0,0 +1,179 @@
+
+struct os_sem sem;
+
+void flash_test_task1(void *arg)
+{
+    const struct flash_area *fa;
+    struct task_cfg *cfg = arg;
+    int i;
+    int increment;
+    uint8_t pat_buf[sizeof(pattern)];
+
+    g_current_task->t_pad = cfg->pin;
+    if (flash_area_open(cfg->flash_area_id, &fa))
+        return;
+
+    while (1) {
+        os_sem_pend(&sem, OS_TIMEOUT_NEVER);
+        console_printf("Task %d starts\n", os_sched_get_current_task()->t_taskid);
+        flash_area_erase(fa, cfg->flash_area_offset, cfg->flash_area_size);
+        increment = cfg->increment;
+        int chunk = 0;
+        for (i = 0 ; i < cfg->flash_area_size; ) {
+            chunk += increment;
+            if (chunk < 0 || chunk > sizeof(pattern)) {
+                increment = -increment;
+                chunk += increment;
+            }
+            if (i + chunk > cfg->flash_area_size)
+                chunk = cfg->flash_area_size - i;
+            flash_area_write(fa, cfg->flash_area_offset + i, pattern, chunk);
+            i += chunk;
+            os_time_delay(1);
+        }
+
+        increment = cfg->increment;
+        chunk = 0;
+        for (i = 0 ; i < cfg->flash_area_size; ) {
+            chunk += increment;
+            if (chunk < 0 || chunk > sizeof(pattern)) {
+                increment = -increment;
+                chunk += increment;
+            }
+            memset(pat_buf, 0xDA, chunk);
+            if (i + chunk > cfg->flash_area_size)
+                chunk = cfg->flash_area_size - i;
+            flash_area_read(fa, cfg->flash_area_offset + i, pat_buf, chunk);
+            if (memcmp(pattern, pat_buf, chunk) != 0) {
+                console_printf("Flast write/read failed\n");
+            }
+            os_time_delay(1);
+            i += chunk;
+        }
+        console_printf("Task %d finished and waits for next start\n",
+                       os_sched_get_current_task()->t_taskid);
+    }
+}
+
+static int spiflash_stress_test_cli_cmd(int argc, char **argv);
+static struct shell_cmd spiflash_stress_cmd_struct = {
+    .sc_cmd = "flashstress",
+    .sc_cmd_func = spiflash_stress_test_cli_cmd
+};
+
+static int
+spiflash_stress_test_cli_cmd(int argc, char **argv)
+{
+    if (argc > 1 && (!strcmp(argv[1], "?") || !strcmp(argv[1], "help"))) {
+        console_printf("Commands Available\n");
+        console_printf("start\n");
+        return 0;
+    }
+
+    os_sem_release(&sem);
+    os_sem_release(&sem);
+    os_sem_release(&sem);
+
+    return 0;
+}
+
+/*
+ * Initialize the package. Only called from sysinit().
+ */
+void
+spiflash_stress_test_init(void)
+{
+    os_sem_init(&sem, 0);
+    runtest_init_task(flash_test_task1, MYNEWT_VAL(OS_MAIN_TASK_PRIO) + 1);
+    runtest_init_task(flash_test_task1, MYNEWT_VAL(OS_MAIN_TASK_PRIO) + 2);
+    runtest_init_task(flash_test_task1, MYNEWT_VAL(OS_MAIN_TASK_PRIO) + 3);
```

Comment:
I think you can use SPIFLASH_STRESS_TEST_TASK_COUNT above as well, that way SPIFLASH_STRESS_TEST_TASK_COUNT can be used to instrument any number of tasks. You can even make it a syscfg then :-)